spec/PeriodicBackgroundSync-index.bs

```diff
@@ -110,3 +131,96 @@ The user agent should cap the duration and frequency of these events to limit re
 Large resources should be downloaded by registering a [=background fetch=] via the {{BackgroundFetchManager}} interface.
 
 In addition, the user agent should consider other factors such as user engagement with the origin, and any user indications to temporarily reduce data consumption, such as a Data Saving Mode, to adjust the frequency of <code>periodicsync</code> events.
+
+Algorithms {#algorithms}
+=========================
+## <dfn>Calculate a time to fire</dfn> ## {#caculate-time-to-fire}
+This section describes how a user agent can calculate the [=periodicsync registration/time to fire=] for a [=periodicsync registration=], |registration|. The time interval between periodicSync events for |registration| MUST be greater than or equal to the {{options/minInterval}} value associated with |registration|.
+
+The user agent MAY include factors such as user engagement with the origin to decide this time interval, allowing origins with high user engagement to update their web apps more often. The user agent SHOULD also ensure this time interval conforms to the caps asserted by [=minimum interval for origin=] and the [=minimum interval across origins=].
+
+The user agent MAY also decide to retry each failed periodicSync event, capped to [=maximum number of retries=].
+
+A possible algorithm to calculate the [=periodicsync registration/time to fire=], |timeToFire| for |registration| would involve running these steps:
+1. If [=count of retries=] for |registration| is 0, <dfn>make a first attempt of firing </dfn> the periodicSync event:
```

Comment:
[=periodicsync registration/count of retries=]

Reply:
done